### PR TITLE
Userland: Allow the Analog Clock window border to be hidden

### PR DESCRIPTION
**Pull request by bgianfo**
A few minor improvements to the new Analog Clock app.
 
-  __Userland: Reduce pledges requested by AnalogClock__

    After startup AnalogClock only needs the normal GUI event loop pledges.

-   __Userland: Allow the Analog Clock window border to be hidden__

    Introduce the ability to hide the window border for the new Analog Clock
    app, with the feature enabled it looks like the clock is floating and
    kind of integrated into the desktop.

    The Cube Demo has the same feature, and was used as inspiration when
    implementing it for the Analog Clock.

cc: @ReasonablePanic 

Example:
unknown

### CODE REVIEW

**Comment by ReasonablePanic** [comment]
**Great idea!** Looks really cool without the background! 
Is there a way to also make it movable while it's border less?

**Comment by bgianfo** [comment]
> Is there a way to also make it movable while it's border less?

I'll poke around a bit more, it wasn't immediately obvious to me when I was looking at the code.

**Comment by ReasonablePanic** [comment]
> > Is there a way to also make it movable while it's border less?
> 
> I'll poke around a bit more, it wasn't immediately obvious to me when I was looking at the code.

Maybe we can add it as a menu option for all windows? the way we do with minimize and maximize.

File: Userland/Applications/AnalogClock/main.cpp [hunk 1]
```diff
@@ -32,12 +34,24 @@ int main(int argc, char** argv)
...
-
-    window->set_main_widget<AnalogClock>();
     window->set_title(Core::DateTime::now().to_string("Clock %d-%m-%Y"));
     window->set_icon(app_icon.bitmap_for_size(16));
     window->resize(170, 170);
     window->set_resizable(false);
+    auto& clock = window->set_main_widget<AnalogClock>();
+
+    // Note: We create the menubar, but never attach it to the actual window,
+    // so it's only reachable via context menu request (i.e. right click).
```
[the remark this answers is not in the document]
Comment (bgianfo) [reply Userland/Applications/AnalogClock/main.cpp]: Ah, thanks!  Done.

**Comment by awesomekling** [comment]
"Intrude the ability"?

**Pull request merged**